p2p/ipld/read_test.go

```diff
@@ -302,6 +302,25 @@ func TestRetrieveBlockData(t *testing.T) {
 	}
 }
 
+func TestPutRetrieveEmptyBlock(t *testing.T) {
+	// create the context and batch needed for node collection from the tree
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// issue a new API object
+	ipfsAPI := mockedIpfsAPI(t)
+
+	block := types.Block{LastCommit: &types.Commit{}}
+	block.Hash()
+
+	err := block.PutBlock(ctx, ipfsAPI.Dag())
+	require.NoError(t, err)
+
+	data, err := RetrieveBlockData(ctx, &block.DataAvailabilityHeader, ipfsAPI, types.DefaultCodec())
+	assert.NoError(t, err)
+	assert.NotNil(t, data)
+}
+
```

Comment:
if we change `generateRandomBlockData` to 
```go
func generateRandomBlockData(msgCount, msgSize int) types.Data {
	var out types.Data
	if msgCount == 0 {
		return out
	}
	out.Messages = generateRandomMessages(msgCount-1, msgSize)
	out.Txs = generateRandomContiguousShares(1)
	return out
}
```
I think we should be able to make this test a test case in `TestRetrieveBlockData`

Reply:
I think it is kinda cool that Hlib added an extra test-case that more explicitly documents this. But you, Evan, are also right that this case (also) belongs to `TestRetrieveBlockData`. Hlib, do you mind integrating this case into the table-driven tests of `TestRetrieveBlockData`? I think it's up to you if you want to keep the `TestPutRetrieveEmptyBlock` (why not).

Reply:
Sure

Reply:
Unluckily, ErasureNMT panics for zero square, but it works for 1